quickwit/quickwit-serve/src/index_api/rest_handler.rs

```diff
@@ -47,75 +51,160 @@ pub fn index_management_handlers(
 }
 
+/// This struct represents the QueryString passed to
+/// the rest API to filter splits.
+#[derive(Debug, Clone, Deserialize, utoipa::IntoParams, utoipa::ToSchema)]
+#[into_params(parameter_in = Query)]
+struct ListSplitsQueryParam {
+    /// A specific split state(s) to filter by.
+    #[serde(deserialize_with = "from_simple_list")]
+    pub split_states: Option<Vec<SplitState>>,
+    /// If set, restrict splits to documents with a `timestamp >= start_timestamp`.
+    pub start_timestamp: Option<i64>,
+    /// If set, restrict splits to documents with a `timestamp < end_timestamp``.
+    pub end_timestamp: Option<i64>,
+    /// If set, restrict splits whose creation dates are before this date.
+    pub end_create_timestamp: Option<i64>,
+}
+
 #[utoipa::path(
     get,
     tag = "Indexes",
     path = "/indexes/{index_id}/splits",
     responses(
-        (status = 200, description = "Successfully fetched all splits.", body = [Split])
+        (status = 200, description = "Successfully fetched splits.", body = [Split])
     ),
     params(
+        ListSplitsQueryParam,
         ("index_id" = String, Path, description = "The index ID to retrieve delete tasks for."),
     )
 )]
-/// Get All Splits
-async fn get_all_splits(
+/// Get splits.
+async fn get_splits(
     index_id: String,
-    index_service: Arc<IndexService>,
-) -> Result<Vec<Split>, IndexServiceError> {
-    info!(index_id = %index_id, "get-all-splits");
-    index_service.get_all_splits(&index_id).await
+    list_split_query: ListSplitsQueryParam,
+    metastore: Arc<dyn Metastore>,
+) -> Result<Vec<Split>, MetastoreError> {
+    info!(index_id = %index_id, list_split_query = ?list_split_query, "get-splits");
+    let mut query = ListSplitsQuery::for_index(&index_id);
+    if let Some(split_states) = list_split_query.split_states {
+        query = query.with_split_states(split_states);
+    }
+    if let Some(start_timestamp) = list_split_query.start_timestamp {
+        query = query.with_time_range_start_gte(start_timestamp);
+    }
+    if let Some(end_timestamp) = list_split_query.end_timestamp {
+        query = query.with_time_range_end_lt(end_timestamp);
+    }
+    if let Some(end_created_timestamp) = list_split_query.end_create_timestamp {
+        query = query.with_create_timestamp_lt(end_created_timestamp);
+    }
+    metastore.list_splits(query).await
 }
 
-fn get_all_splits_handler(
-    index_service: Arc<IndexService>,
+fn get_splits_handler(
+    metastore: Arc<dyn Metastore>,
 ) -> impl Filter<Extract = (impl warp::Reply,), Error = Rejection> + Clone {
     warp::path!("indexes" / String / "splits")
         .and(warp::get())
-        .and(with_arg(index_service))
-        .then(get_all_splits)
+        .and(serde_qs::warp::query(serde_qs::Config::default()))
+        .and(with_arg(metastore))
+        .then(get_splits)
+        .map(format_response)
+}
+
+#[derive(Deserialize)]
+#[serde(deny_unknown_fields)]
+struct SplitsForDeletion {
```

Comment:
I guess using `SplitsForDeletion` instead of  `Vec<String>` is for consistency right?

Reply:
The rationale behind is that I want to have a body that is not directly an array, it's more a convention that anything else but I like to have explicit keys and values and I decode a body into json.